### ISSUE TITLE
Implement 'P1285R0 Improving Completeness Requirements for Type Traits'

### ISSUE THREAD

**Issue by cjdb** [llvm/llvm-project#99995]
A robust implementation of P1285 should render a program ill-formed if a program breaks the rules outlined in the paper. This will require some front-end support.

**Comment by cor3ntin** [comment]
Yeah... I am not sure there would be something actually implementable here - which is why it isn't outright ill-formed. 

For example https://compiler-explorer.com/z/PxnxxbnhT - all of it is memoized and trying to re-instantiate everything to see whether the result changes doesn't sound practical or desirable (e.g, if we could, the impact on compile times would not please any one)
